kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -114,6 +114,15 @@ public TopicPartitionWriter(TopicPartition tp,
     flushSize = connectorConfig.getInt(S3SinkConnectorConfig.FLUSH_SIZE_CONFIG);
     topicsDir = connectorConfig.getString(StorageCommonConfig.TOPICS_DIR_CONFIG);
     rotateIntervalMs = connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_INTERVAL_MS_CONFIG);
+    if (rotateIntervalMs > 0) {
+      log.warn(
+          "Property '{}' is set to '{}ms' but partitioner is not an instance of {}. This property"
+              + " is ignored.",
+          S3SinkConnectorConfig.ROTATE_INTERVAL_MS_CONFIG,
+          rotateIntervalMs,
+          TimeBasedPartitioner.class.getName()
```

Comment:
how do we know that `partitioner` is not an instance of `TimeBasedPartitioner ` here?

Reply:
Good call. I meant to use the `timestampExtractor` initialized above, but I didn't.